GeneratorInterface/Core/src/BaseHadronizer.cc

```diff
@@ -1,20 +1,154 @@
+        if (randomizedParameters[irand].exists("ConfigDescription")) {
+          randomInitConfigDescriptions_[irand] = randomizedParameters[irand].getParameter<std::string>("ConfigDescription");
+        }
+        if (randomizedParameters[irand].exists("GridpackPath")) {
+          gridpackPaths_[irand] = randomizedParameters[irand].getParameter<std::string>("GridpackPath");
+        }
+      }
+    }
+    else {
+      if (ps.exists("GridpackPath")) {
+        gridpackPaths_[0] = ps.getParameter<std::string>("GridpackPath");
+      }
+    }
+    
+    runInfo().setFilterEfficiency(
+        ps.getUntrackedParameter<double>("filterEfficiency", -1.) );
+    runInfo().setExternalXSecLO(
+        GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSection", -1.)) );
+    runInfo().setExternalXSecNLO(
+        GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSectionNLO", -1.)) );
 
-   runInfo().setFilterEfficiency(
-      ps.getUntrackedParameter<double>("filterEfficiency", -1.) );
-   runInfo().setExternalXSecLO(
-      GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSection", -1.)) );
-   runInfo().setExternalXSecNLO(
-       GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSectionNLO", -1.)) );
+  }
+  
+  GenLumiInfoHeader *BaseHadronizer::getGenLumiInfoHeader() const {
+    
+    GenLumiInfoHeader *genLumiInfoHeader = new GenLumiInfoHeader();
+    
+    //fill information on randomized configs for parameter scans
+    genLumiInfoHeader->setRandomConfigIndex(randomIndex_);
+    if (randomIndex_>=0) {
+      genLumiInfoHeader->setConfigDescription(randomInitConfigDescription());      
+    }
+    
+    return genLumiInfoHeader;
+    
+  }
 
-}
+  void BaseHadronizer::randomizeIndex(edm::LuminosityBlock const& lumi, CLHEP::HepRandomEngine* rengine) {
+    if (randomInitWeights_.size()>0) {
+      //randomly select from a list of provided configuration sets (for parameter scans)
+      
+      //seeds std 32-bit mersene twister with HepRandomEngine state plus run and lumi section numbers
+      //(random engine state will be the same for every lumi section in a job)
+      std::vector<long unsigned int> seeds = rengine->put();
+      seeds.push_back(lumi.id().run());
+      seeds.push_back(lumi.id().luminosityBlock());
+      std::seed_seq seedseq(seeds.begin(),seeds.end());
+      std::mt19937 randgen(seedseq);
+      std::discrete_distribution<int> randdist(randomInitWeights_.begin(),randomInitWeights_.end());
+      
+      randomIndex_ = randdist(randgen);
+      
+      printf("selected random index = %i\n",randomIndex_);
+      
+    }
+  }
+  
+  void BaseHadronizer::generateLHE(edm::LuminosityBlock const& lumi, CLHEP::HepRandomEngine* rengine) {
+        
+    if (gridpackPath().empty()) {
+      return;
+    }
+    
+    //get random seed from HepRandomEngine state plus run and lumi section numbers
```

Comment:
and another variant of the random number initialization here where in this case no random number generator is seeded within the code, but the HepJamesRandom state is used with run and lumi numbers to generate a new (single) seed for the gridpack lhe generation script.